internal/services/springcloud/spring_cloud_container_deployment_resource.go

```diff
@@ -320,3 +336,32 @@ func expandSpringCloudContainerDeploymentResourceRequests(input []interface{}) *
 
 	return &result
 }
+
+func expandSpringCloudDeploymentApms(input []interface{}) *[]appplatform.ApmReference {
+	if len(input) == 0 {
+		return nil
+	}
+	result := make([]appplatform.ApmReference, 0)
+	for _, v := range input {
+		result = append(result, appplatform.ApmReference{
+			ResourceID: pointer.To(v.(string)),
+		})
+	}
+	return pointer.To(result)
+}
+
+func flattenSpringCloudDeploymentApms(input *[]appplatform.ApmReference) []interface{} {
+	if input == nil {
+		return nil
+	}
+	result := make([]interface{}, 0)
+	for _, v := range *input {
+		id, err := appplatform2.ParseApmIDInsensitively(*v.ResourceID)
+		if err != nil {
+			log.Printf("[WARN] invalid APM ID %q: %+v", *v.ResourceID, err)
+			continue
```

Comment:
We should be throwing an error here if the resource ID can't be parsed since this would indicate a problem in the API that the service team should fix.

If we silently log a warning and continue we would be allowing malformed resource ID's to creep into the state which will cause problems further down the line if user's try to use these for something else.

Reply:
Got it! I've updated this PR, please take another look, thanks!